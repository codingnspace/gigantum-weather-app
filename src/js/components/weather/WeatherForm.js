//Vendor
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
//Locals
import temperature from 'images/temperature.svg';
import percent from 'images/percent.svg';

class WeatherForm extends Component {
  static propTypes = {
    handleSubmit: PropTypes.func,
    getWheather: PropTypes.func
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.handleSubmit((formValues) => {
      this.props.getWeather(formValues);
    })();
  }

  render () {
    return (
      <div className="weather-form-container">
        <h2>Put In Your Conditions</h2>
        <form onSubmit={this.handleSubmit} className="weather-form">
            <div className="input-with-icon">
              <Field component="input" placeholder="Min Temp" name="minTemp" />
              <img className="app-logzzo" src={temperature} alt="app-logo" />
            </div>
            <div className="input-with-icon">
              <Field component="input" placeholder="Max Temp" name="maxTemp" />
              <img className="app-logzzo" src={temperature} alt="app-logo" />
            </div>
            <div className="input-with-icon">
              <Field component="input" placeholder="Max Rain Percentage" name="maxRain" />
              <img className="app-logzzo" src={percent} alt="app-logo" />
            </div>
            <div className="btn-wrapper">
              <button type="submit">Get Info</button>
            </div>
        </form>
      </div>
    );
  }
}

export default reduxForm({form: 'weather'})(WeatherForm);
