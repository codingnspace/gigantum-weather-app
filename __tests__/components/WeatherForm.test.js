import { shallowToJson } from 'enzyme-to-json';
import { configure, shallow } from 'enzyme';
import React from 'react';

import WeatherForm from 'js/components/weather/WeatherForm';
import Adapter from 'enzyme-adapter-react-15';

configure({adapter: new Adapter()});

describe('WeatherForm', () => {
  it('should render correctly', () => {
    const output = shallow(
      <WeatherForm getWeather={()=>null} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
