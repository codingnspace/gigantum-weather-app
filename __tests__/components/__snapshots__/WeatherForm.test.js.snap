// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherForm should render correctly 1`] = `
<Connect(Form(WeatherForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="weather"
  getFormState={[Function]}
  getWeather={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
